resolwe_bio/processes/import_data/differential_exp.yml

```diff
@@ -114,24 +113,166 @@
       if [[ '.{{ src.file }}' =~ \.(xls)$ ]]; then
         re-import "{{ src.file_temp|default:src.file }}" "{{ src.file }}" "xls" "xls" 0.3 extract
         convert_DE_excel_table.py '{{ src.file }}' > "${NAME}.tab"
-        parse_diffexp.py ${NAME}.tab ${GENE_ID} ${LOG_FC} ${FDR} ${LOGODDS} ${FWER} ${PVALUE} ${STAT}
+        parse_diffexp.py ${NAME}.tab ${GENE_ID} ${LOGFC} ${FDR} ${LOGODDS} ${FWER} ${PVALUE} ${STAT} --output de_data.json
         re-checkrc
         re-progress 0.7
         gzip "${NAME}.tab"
       elif [[ '.{{ src.file }}' =~ \.(xlsx)$ ]]; then
         re-import "{{ src.file_temp|default:src.file }}" "{{ src.file }}" "xlsx" "xlsx" 0.3 extract
         convert_DE_excel_table.py '{{ src.file }}' > "${NAME}.tab"
-        parse_diffexp.py ${NAME}.tab ${GENE_ID} ${LOG_FC} ${FDR} ${LOGODDS} ${FWER} ${PVALUE} ${STAT}
+        parse_diffexp.py ${NAME}.tab ${GENE_ID} ${LOGFC} ${FDR} ${LOGODDS} ${FWER} ${PVALUE} ${STAT} --output de_data.json
         re-checkrc
         re-progress 0.7
         gzip "${NAME}.tab"
       else
         re-import "{{ src.file_temp|default:src.file }}" "{{ src.file }}" "diff|tab|gz" "tab" 0.3
-        parse_diffexp.py ${NAME}.tab ${GENE_ID} ${LOG_FC} ${FDR} ${LOGODDS} ${FWER} ${PVALUE} ${STAT}
+        parse_diffexp.py ${NAME}.tab ${GENE_ID} ${LOGFC} ${FDR} ${LOGODDS} ${FWER} ${PVALUE} ${STAT} --output de_data.json
+        re-checkrc
+        re-progress 0.7
+      fi
+      re-save-file raw ${NAME}.tab.gz
+      re-save-file de_file de_file.tab.gz
+      re-save de_json de_data.json
+      re-save source "{{source}}"
+
+- slug: upload-diffexp-inputs
```

Comment:
Refactor the `upload-diffexp` process and remove the `upload-diffexp-inputs`. Add `case` and `control` fields as optional fields to the `upload-diffexp` process. This way you will not have to repeat the process logic and bioinformaticians will only have to remember one slug.